### PR TITLE
Add GUI for teams feature, update UG and DG.

### PR DESCRIPTION
**Pull request by angsongyee**
## GUI changes
- Team list added to GUI beside the contact list
- The minimum window size has been increased to ensure that all UI
  components are not clipped.
- All top level containers (command input, result window,
  contact/team list) now expand to fill the available space instead
  of having a fixed width/height
- Contact's tags/techstack will now overflow to a second row instead
  being truncated by the contact card's width
- Font sizes for command input and result window have been increased
  for better readability.
- Default profile picture now uses a locally saved image instead of loading the image from the web.
## Tasks
- [x] Add GUI for teams feature 
- [x] Update UG and DG

### CODE REVIEW

**Review by yuechen2001 — APPROVED** [review]
Not a GUI expert so i trust u

**Comment by angsongyee** [comment]
### Updated UI Class Diagram:
image

### Updated Model Class Diagram:
image

### New Team Class Diagram:
image

File: docs/UserGuide.md [hunk 1]
```diff
@@ -248,6 +248,66 @@ Exits the program.
...
+
+- `team 1` updates the contact list to show the members of the first team.
+
+#### Deleting a team: `delete`
+
+You can use the `delete` team command to delete a team from your team list. After deleting a team, you will still be able to find its members in the contact list.
+
+Format: `team TEAM_INDEX delete`
+
+- Export the members' details of the team at the specified `TEAM_INDEX`.
```
Comment (maiyasaliha) [inline docs/UserGuide.md:289]:
```suggestion
- Deletes the team at the specified `TEAM_INDEX`.
```

File: docs/UserGuide.md [hunk 2]
```diff
@@ -248,6 +248,66 @@ Exits the program.
...
+- The index refers to the index number shown in the displayed teams list.
+- The index **must be a positive integer** 1, 2, 3, …​
+
+Examples:
+
+- `team 1 export` deletes the first team in the team list.
```
Comment (maiyasaliha) [inline docs/UserGuide.md:295]:
```suggestion
- `team 1 delete` deletes the first team in the team list.
```

File: src/main/resources/view/ContactListCard.fxml [hunk 3]
```diff
@@ -1,50 +1,47 @@
...
-<?import javafx.scene.layout.HBox?>
-<?import javafx.scene.layout.Region?>
-<?import javafx.scene.layout.VBox?>
-<?import javafx.scene.image.ImageView?>
-<?import javafx.scene.layout.RowConstraints?>
+
+<?import javafx.geometry.*?>
+<?import javafx.scene.control.*?>
+<?import javafx.scene.image.*?>
+<?import javafx.scene.layout.*?>
```
Comment (maiyasaliha) [inline src/main/resources/view/ContactListCard.fxml:6]: Wildcard Imports

File: src/main/resources/view/MainWindow.fxml [hunk 4]
```diff
@@ -1,60 +1,62 @@
...
-<?import javafx.scene.image.Image?>
-<?import javafx.scene.layout.StackPane?>
-<?import javafx.scene.layout.VBox?>
+<?import java.net.*?>
+<?import javafx.geometry.*?>
+<?import javafx.scene.*?>
+<?import javafx.scene.control.*?>
+<?import javafx.scene.image.*?>
+<?import javafx.scene.layout.*?>
+<?import javafx.stage.*?>
```
Comment (maiyasaliha) [inline src/main/resources/view/MainWindow.fxml:9]: Wildcard Imports

**Review by maiyasaliha — CHANGES_REQUESTED** [review]
LGTM! Just some minor changes to be addressed before merging. Also take note of the arrowhead in fxml files, it is not huge, and honestly idk if we can avoid it in fxml. Just highlighting for code quality.

**Review by yuechen2001 — APPROVED** [review]

**Comment by angsongyee** [comment]
Normal width:
image
Narrow width:
image

File: docs/UserGuide.md [hunk 5]
```diff
@@ -249,17 +249,59 @@ Exits the program.
...
-All teams related commands start with the prefix: `team`
+All team related commands start with the prefix: `team`
+
+#### Adding a team: `add`
+
+You can use the `add` team command to create teams of contacts, which will help you to keep track of your teams when joining hackathons!
+
+  Format: `team add n/TEAM_NAME`
+
+- TEAM_NAME can only contain alphanumeric characters and spaces and cannot be blank.
```
Comment (Teee728) [inline docs/UserGuide.md:262]: minor change, can do `TEAM_NAME` instead

**Review by Teee728 — APPROVED** [review]
LGTM!

**Pull request merged**